File: MainServer/src/main/java/Domain/UserData/Invitation.java
package Domain.UserData;

import Domain.UserData.Interfaces.IInvitation;

import java.time.LocalDateTime;

public class Invitation implements IInvitation {

    private String id;
    private String teamId;
    private String swimmerId;
    private LocalDateTime creationTime;
    protected InvitationStatus status;

    public Invitation(String id, String teamId, String swimmerId) {
        this.id = id;
        this.teamId = teamId;
        this.swimmerId = swimmerId;
        this.creationTime = LocalDateTime.now();
        this.status = InvitationStatus.PENDING;
    }

    public Invitation(String id, String teamId, String swimmerId, LocalDateTime time, InvitationStatus status) {
        this.id = id;
        this.teamId = teamId;
        this.swimmerId = swimmerId;
        this.creationTime = time;
        this.status = status;
    }

    public Invitation(Invitation other) {
        this.id = other.id;
        this.teamId = other.teamId;
        this.swimmerId = other.swimmerId;
        this.creationTime = other.creationTime;
        this.status = other.status;
    }

    public String getId() {
        return id;
    }

    public String getTeamId() {
        return teamId;
    }

    public String getSwimmerId() {
        return swimmerId;
    }

    public LocalDateTime getCreationTime() {
        return creationTime;
    }

    @Override
    public boolean isPending() {
        return status == InvitationStatus.PENDING;
    }

    @Override
    public boolean isApprove() {
        return status == InvitationStatus.ACCEPTED;
    }

    @Override
    public boolean isDenied() {
        return status == InvitationStatus.DENY;
    }

    public InvitationStatus getStatus() {
        return status;
    }
}

File: client/lib/Domain/Invitations/Invitation.dart
import 'package:client/Domain/Dates/DateDayDTO.dart';

class Invitation {

  String id;
  String teamId;
  String email;
  DateDayDTO date;
  bool isPending;
  bool isApprove;
  bool isDenied;


  Invitation(this.id, this.teamId, this.email, this.date, this.isPending,
      this.isApprove, this.isDenied);

  Invitation.fromJson(Map<String, dynamic> map)
    : id = map['id'],
      teamId = map['teamId'],
      email = map['email'],
      date = DateDayDTO.fromJson(map['dateDayDTO']),
      isPending = map['isPending'],
      isApprove = map['isApprove'],
      isDenied = map['isDenied'];


  int compareTo(Invitation other) {
    if(isApprove && !other.isApprove) {
      return -1;
    }
    else if(!isApprove && other.isApprove) {
      return 1;
    }
    else {
      if(isPending && !other.isPending) {
        return -1;
      }
      else if(!isPending && other.isPending) {
        return 1;
      }
      else {
        if(isDenied && !other.isDenied) {
          return -1;
        }
        else if(!isDenied && other.isDenied) {
          return 1;
        }
      }
    }
    return 0;
  }

}